### DISCUSSION TITLE
Number Highlight customization options for lsp

### DISCUSSION THREAD

**Discussion by saifulapm** [LazyVim/LazyVim#108] [Ideas]
Hi, 

if Lazyvim can provide  ```numhl = hl``` option in lsp opts, It will be awesome. I know that I can overwrite whole lsp config but I always love to use Lazyvim config becuase I don't want to miss future updates. 

```lua
 vim.fn.sign_define(name, { text = icon, texthl = name, numhl = "" })
```

I alwasy use numhl instead of vertual text becuase I love minimal look. 

Anyway, Thanks @folke for your awesome work on neovim community.

**Comment by folke** [comment]
What does `numhl=false` do? The docs don't mention this. Isn't the end-result the same as `numhl=""`?

**Comment by folke** [comment]
Oh, or do you want to skip the `numhl=""`?

**Comment by folke** [comment]
What should it be instead then?

**Comment by saifulapm** [comment]
Hi, I use ```numbhl```. It highlight that number based on the diagnostic. 

```lua

local sign = require("const.DIAGNOSTIC_SIGN")
  local severities = {
    [vim.diagnostic.severity.ERROR] = { text = "Error", sign = sign.ERROR },
    [vim.diagnostic.severity.WARN] = { text = "Warn", sign = sign.WARN },
    [vim.diagnostic.severity.INFO] = { text = "Info", sign = sign.INFO },
    [vim.diagnostic.severity.HINT] = { text = "Hint", sign = sign.HINT },
  }

  for _, severity in pairs(severities) do
    local hl = ("DiagnosticSign%s"):format(severity.text)
    vim.fn.sign_define(hl, { text = severity.sign, texthl = hl, numhl = hl })
  end
```

This is current config.

**Comment by saifulapm** [comment]
i wonder if i can do this on lazyvim without replacing whole config. Thanks

**Comment by saifulapm** [comment]
current lazyvim config like this:

```lua
vim.fn.sign_define(name, { text = icon, texthl = name, numhl = "" })
```

I want to make it like
```lua
vim.fn.sign_define(name, { text = icon, texthl = name, numhl = name })
```

But problem everyone will not want to highlight number so there is extra setting needed.

**Comment by folke** [comment]
I thought about it, but I don't think its a good idea to make this configurable.
You can add the below to your specs though:

```lua
  {
    "neovim/nvim-lspconfig",
    config = function(plugin, opts)
      plugin._.super.config(plugin, opts)
      for name, icon in pairs(require("lazyvim.config").icons.diagnostics) do
        name = "DiagnosticSign" .. name
        vim.fn.sign_define(name, { text = icon, texthl = name, numhl = name })
      end
    end,
  },
```

**Reply by saifulapm** [reply]
Thank you

**Reply by saifulapm** [reply]
```plugin._.super.config(plugin, opts)```

Is there any issue with this line ? after doing like this, lsp not attching. 
```
config = function(plugin, opts)
      plugin._.super.config(plugin, opts)
      for name, icon in pairs(require("lazyvim.config").icons.diagnostics) do
        name = "DiagnosticSign" .. name
        vim.fn.sign_define(name, { text = icon, texthl = name, numhl = name })
      end
    end,
```

**Reply by folke** [reply]
You should just add the code block I mentioned above to one of your spec files like `lua/plugins/test.lua`. All lsp functionality should work as expected.
